LayoutTests/platform/mac-site-isolation/TestExpectations

```diff
@@ -500,6 +503,16 @@ imported/w3c/web-platform-tests/html/browsers/windows/targeting-cross-origin-nes
 ########################################################################
 # Tests that run to completion but indicate differences we need to fix #
 ########################################################################
+accessibility/datalist.html [ Pass ImageOnlyFailure ]
+accessibility/img-alt-tag-only-whitespace.html [ Pass ImageOnlyFailure ]
+accessibility/mac/basic-embed-pdf-accessibility.html [ ImageOnlyFailure ]
+accessibility/mac/iframe-pdf.html [ ImageOnlyFailure ]
+accessibility/mac/media-role-descriptions.html [ Pass ImageOnlyFailure ]
+accessibility/mac/scrolling-in-pdf-crash.html [ Pass ImageOnlyFailure ]
+accessibility/mac/webkit-scrollarea-position.html [ Pass ImageOnlyFailure ]
+accessibility/model-element-attributes.html [ Pass ImageOnlyFailure ]
+accessibility/progressbar.html [ Pass ImageOnlyFailure ]
+accessibility/tree-update-with-dirty-layout.html [ Pass ImageOnlyFailure ]
```

Comment:
I'm not sure if it's been done correctly, but the others in this category are [ ImageOnlyFailure Failure ].  Does [ Pass ImageOnlyFailure ] make a test run indicate that there is no issue with these tests?

Reply:
Hmm, well spotted!
I copied the results from `run-webkit-tests --iterations=10 ...`. I assumed this meant intermittent failures, so each run would either Pass, or fail with exactly "ImageOnlyFailure".

Indeed, according to https://trac.webkit.org/wiki/TestExpectations, "If multiple expectations are listed, the test is considered "flaky" and any of those results will be considered as expected."
So `[ ImageOnlyFailure Failure ]` should therefore mean a test always fails with either "ImageOnlyFailure" or a plain "Failure".

Focusing on the first test, accessibility/datalist.html, if I run it with `--iterations=10`, I can see it failing about half the time with "ImageOnlyFailure", and it shows the result "accessibility/datalist.html [ ImageOnlyFailure Pass ]", which makes sense to me.

I then tried to run it a few times without `--iterations`, and either it passed (as half-expected), or it failed; but it still reported an unexpected failure with result "accessibility/datalist.html [ ImageOnlyFailure ]" -- I would have expected an *expected* failure there!?

I also tried in mac-site-isolation/TestExpectations: `[ ImageOnlyFailure ]`, and `[ ImageOnlyFailure Failure ]`, but it behaved the same way, still complaining about unexpected failures, and *not* complaining about passes (even though "Pass" is not in the list of expectations).

IIRC you told me that `--site-isolation` runs the test twice (with and without site-isolation), and compares the results? I'll try to have a look at how that works, and why it still complains about results that are in the TestExpectations file, and doesn't complain about passes when "Pass" is not expected...
Please let me know if you have other ideas.